topics-and-advance-readings/psdad.md

```diff
@@ -0,0 +1,213 @@
+
+## Alternatives
+
+I am not aware of anyone else having done work on this problem. I have
+developed another solution which applies only to the RDF space,
+[Movable Schemas](https://sandhawke.github.io/mov/), possibly with
+[version-integrity](https://github.com/sandhawke/version-integrity). PSDAD
+is the extension of Movable Schemas idea to the larger field beyond
+RDF.
+
+## Proposal
+
+PSDAD starts with the observation that the semantics of each field in
+a data format are in practice defined by a short paragraph in some
+specification. In the example above, Alice and Bob might be using a
+spec which contains this text:
+
+> ...
+>
+> **probe**: The "probe" field specifies the integer index of the probe whose temperature is specified in the "temp" field.
+>
+> **temp**: The "temp" field specifies the temperature measured at the given probe, rounded off to the nearest integer, in degrees Celsius.
+>
+> ...
+
+We want that spec text to be connected to the bits on the wire, in a
+way that is secure, scalable, and maintainable. Operationally, we want
+the people writing any software which might be working with the data
+in those fields to be looking at the same spec text, with minimal
+chance for error or disinformation.
+
+The PSDAD approach is to send that spec text on the wire wrapped
+around the data fields.  For example, Alice might transmit:
+
+```text
+The temperature at probe number 6 is 34°C.
+```
+
+To a human who knows English, this is fairly meaningful.  To the
+software at either end, this looks pretty much the same as the JSON
+expression given above, `{"probe":6,"temp": 34}`.  It's a few bytes
+longer, but it's still "stuff 6 stuff 34 stuff".
+
+The PSDAD trick is to make the delimiters be the documentation by
+writing the documentation as template assertions, then filling in the
+templates with the field data values.
+
+The template in use here is `The temperature at probe number [] is
+[]°C.` Given that template, software at either end can easily
+serialize and deserialize probe temperature data.
+
+## Applicability
+
+The primary use case here is to have decentralized extensibility be
+robust against disinformation attacks, but this technique also has
+potential to address schema and documentation management issues in
+conventional deployment environments. In particular, PSDAD allows
+developers to publish and/or consume machine-readable data with semantic
+clarity and minimal procedural work or need for prior agreement.
+
+If you want to publish some data, just think about how you would
+express each item in that data clearly, pedantically, as a statement
+in your preferred natural language.  If you want to consume some data,
+look at some examples and deduce the template.  If it's not obvious,
+there's probably an issue.
+
+Here's another example.  Alice wants to help build a web of trust, so
+she publish this text at https://alice.example/endorsements :
+
+```text
+I have met and know the person "Susan X. Lastname".
```

Comment:
So, this is really interesting because:

1. It biases towards a particular language in a negative way.
2. It biases towards a particular language in a positive way.

The negative way it biases is that it may be that English is used for popular vocabularies and that biases towards Western mental models.

The positive way it biases is that some concepts can't be expressed in some languages and so using the native language to express the concept is very useful.

Reply:
This is most likely a really fringe use case, but I found it interesting.